scripts/scheduled_posts.coffee

```diff
@@ -15,21 +15,18 @@ switch process.env.NODE_ENV
 # Connect to database
 db = mongojs(process.env.MONGOHQ_URL, ['articles'])
 
-db.articles.update(
-  { scheduled_publish_at: { $lt: new Date() } }
-  { 
-    $set: {
-      published: true
-      published_at: Date()
-      scheduled_publish_at: null 
-    } 
-  }
-  { multi: true }
-  (err, msg) ->
-    return exit err if err
-    debug 'Scheduled publication finished', msg
+db.articles.find(
+  scheduled_publish_at: { $lt: new Date() }
+).forEach (err, doc) ->
+  if !doc
+    debug 'Scheduled publication finished'
     process.exit()
-)
+  if err
+    exit err
+  doc.published = true
+  doc.published_at = moment(doc.scheduled_publish_at).toDate()
+  doc.scheduled_publish_at = null
+  db.articles.save doc
```

Comment:
:-o that's odd that it won't typecast using `$set` but will using `save` that seems like a bug in mongojs or the Node mongodb driver. Given the latter issue though, I think this is a great way to kill two birds with one stone.

Solid! :moyai: :boom: :bird: :bird: :dash: :ghost: :ghost: